### DISCUSSION TITLE
How to get the index corresponding to the maximum for a resampled array?

### DISCUSSION THREAD

**Discussion by AndrewILWilliams** [pydata/xarray#5117] [Q&A]
Hi there,

I have a `DataArray` of precipitation with `time, lat, lon` coordinates, and I would like to find the time index which corresponds to the maximum precipitation value *in a given year*.

```python3
>>> print(da_hist)

<xarray.DataArray 'pr' (time: 56978, lat: 64, lon: 128)>
dask.array<open_dataset-aac2e5d3ea1a310f230a62870ced1941pr, shape=(56978, 64, 128), dtype=float32, chunksize=(56978, 64, 128), chunktype=numpy.ndarray>
Coordinates:
  * time     (time) datetime64[ns] 1850-01-01T12:00:00 ... 2005-12-31T12:00:00
  * lat      (lat) float64 -87.86 -85.1 -82.31 -79.53 ... 79.53 82.31 85.1 87.86
  * lon      (lon) float64 0.0 2.812 5.625 8.438 ... 348.8 351.6 354.4 357.2
Attributes:
    standard_name:     precipitation_flux
    long_name:         Precipitation
    comment:           at surface; includes both liquid and solid phases from...
    units:             kg m-2 s-1
    original_name:     PRCP
    original_units:    kg/m**2/s
    history:           2011-10-04T11:32:54Z altered by CMOR: Converted units ...
    cell_methods:      time: mean
    cell_measures:     area: areacella
    associated_files:  baseURL: http://cmip-pcmdi.llnl.gov/CMIP5/dataLocation...
```

I thought that perhaps I could use `da_hist.resample(time='1Y').argmax('time')`, but it seems that `argmax` hasn't been implemented for resampled objects yet?

```python3
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-71-6c5d305923f9> in <module>
----> 1 da_hist.resample(time='1Y').argmax('time').compute()

AttributeError: 'DataArrayResample' object has no attribute 'argmax'
```

Does anyone know a good way around this? Thanks!

**Comment by AndrewILWilliams** [comment]
Follow up, I'd also like to be able to do this for different month groupings as well. i.e., what is the time-index corresponding to the maximum precipitation value within May-September (MJJAS) in each year.

I think I should be able to do this with some clever masking, but I'd still require to be able to use `resample().argmax()`...

```python3
mjjas_mask = (da_hist.time.dt.month >= 5) & (da_hist.time.dt.month <= 9)
da_hist.where(mjjas_mask).resample(time='1Y').argmax('time').compute()
```

**Reply by dcherian** [reply]
Yes, this is a regression. We'll have to add `arg*` to the `GroupBy` class i guess. https://github.com/pydata/xarray/issues/4476

**Reply by AndrewILWilliams** [reply]
Cool! I might have time to take a look later this month depending on how the PhD is going..

**Comment by AndrewILWilliams** [comment]
Ok, I'm confused... it seems like you USED to be able to do this in [v0.15.1](http://xarray.pydata.org/en/v0.15.1/generated/xarray.core.resample.DataArrayResample.argmax.html?highlight=argmax), but you can't now?

**Reply by keewis** [reply]
yes, we did a refactor of `argmin` / `argmax`, but the implementations for `GroupBy` / `Resample` objects (probably also `Coarsen`) are still missing. PRs welcome.

**Reply by DamienIrving** [reply]
Is there a work around to achieve the old `xarray.core.resample.DataArrayResample.argmax` functionality in the meantime? Perhaps passing `numpy.argmax` to `xarray.core.resample.DatasetResample.reduce` would work?
